### DISCUSSION TITLE
Milvus v0.11.0 release not supported anymore

### DISCUSSION THREAD

**Discussion by maximilianbiber** [milvus-io/milvus#4793] [Q&A and General discussion] [answered]
I am currently using Milvus v0.11.0 (docs: https://www.milvus.io/docs/v0.11.0/overview.md) with pymilvus 0.3.0 and just saw that the release is not supported anymore, i.e. the latest release is now v.0.10.6.

With this, one key feature is not supported anymore: "Scalar field filtering".

I couldn't find anything regarding the release change and why v0.11.0 is not supported anymore. 
Is there any way that the release including the "Scalar field filtering" feature is supported again at some point? Also, is it planned to support string types for inserting and filtering (as stated in "Insert Entities" https://pymilvus.readthedocs.io/en/0.3.0/tutorial.html#insert-entities)?

Thanks in advance!

**Comment by yhmo** [comment, marked as answer]
Yes, we stop maintainance for 0.11.0 and rename it to 'experimental'. The main reason is we want to make milvus be a real distribute and cloud-native service. The old architecture can works well as a stand-alone service or small sharding system,  but not good enough to be a real distribute system.

Now we are developing a new milvus with new atchitecture and new language(golang), we plan to publish it at the end of the month, also in open source way. The new features of 0.11.0 will be considered and designed in the new milvus, But not all features are available at the first version, they will be implemented step by step  in later iteration.

**Reply by maximilianbiber** [reply]
Thanks a lot for the quick response! It is great to hear that the new features of 0.11.0 will be considered in the new milvus.
I still have some open questions though:

(1) What exactly is meant by "real distribute and cloud-native service"? Will Milvus just be supported with a cloud connection or can it still be used without any cloud services?

(2) Can you give an estimation about the robustness of the current v0.11.0? Which features are not sophisticated enough and were the reason to rename it to "experimental"?

(3) How will the new architecture differ from the one in v0.11.0?

(4) Since you are releasing the features step-by-step: can you give an estimate on when the "Scalar field filtering" will be supported again?

Thanks a lot for your support!

**Reply by yhmo** [reply]
(1) The new milvus mainly face to cloud service, but we also provide stand-alone milvus with same source code. But the backend meta service is different, the old milvus requires MySQL, while the new milvus uses ETCD.
(2) The v0.11.0 can pass all regression test cases, but it still has some critical, for example: #4684. If we continue maintain it, the bug can be fixed definitely. But now, we have to put resource in new milvus and make the 0.10.x be more stable. It is a hard decision.
(3) The new milvus is written in Golang, most of sdk api is similar to the old milvus, and basic concept is same: segment/partition/collection/index. The difference is the storage format, cache machinery, and segment doesn't need merge in the new milvus.
(4) Current target is to make the new milvus pass main function test cases, some leaders are writing the documentation. The first step is to public it at the end of this month. So far as I know, the "Scalar field filtering" is code ready, but not tested yet.

**Reply by sworddish** [reply]
@yhmo So if v0.10.6 or v0.11.0 is being used in production by some company, and they need an in-house stand-alone service, after new golang version released, they have to do large scale data migration by themself if they want switch to new version? or they have to find alternative solution which means abandoning Milvus Cloud Service in the near future.

**Reply by yhmo** [reply]
> @yhmo So if v0.10.6 or v0.11.0 is being used in production by some company, and they need an in-house stand-alone service, after new golang version released, they have to do large scale data migration by themself if they want switch to new version? or they have to find alternative solution which means abandoning Milvus Cloud Service in the near future.

Firstly we suggest don't use 0.11.0 in production.
Secondly, the new milvus cannot compatible old milvus data. So if user has put 0.10.x to production, he has three choices:
1. continue use 0.10.0(1.0), we will maintain it. we will fix critical issues, but big feature could not be done 
2. upgrade to new milvus once it become stable, we can provide data extract/migration tool for users. Personally I think to make the new milvus to be stable, we still have a lot of work to do.
3. find alternative solution.

**Reply by yhmo** [reply]
> @yhmo So if v0.10.6 or v0.11.0 is being used in production by some company, and they need an in-house stand-alone service, after new golang version released, they have to do large scale data migration by themself if they want switch to new version? or they have to find alternative solution which means abandoning Milvus Cloud Service in the near future.

Yes. 
We will try provide help to users for migration data if they want to upgrade to new milvus.

**Comment by michellemay** [comment]
I'm glad you asked this question.. Filtering is something mandatory on our side and without it, we cannot use milvus..

**Reply by yhmo** [reply]
The new milvus will include this feature.
